src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -159,55 +179,61 @@ def _try_parse_key_value_object(parameters, template_obj, value):
     return True
 
 
-def _process_parameters(template_obj, parameter_lists):  # pylint: disable=too-many-statements
-
-    def _try_parse_json_object(value):
+def _process_parameters_and_ext_configs(template_obj, parameter_lists):  # pylint: disable=too-many-statements
```

Comment:
Personally I think we should avoid overloading `--parameters` any further - the logic is already extremely complicated, and the original goal was to explicitly introduce `extensionConfigs` as a distinct feature.

If it simplifies implementation, I would consider cutting scope by blocking scenarios we don't want to support - for example, if someone is using a `.bicepparam` file, I don't think we need to allow them to override `extensionConfigs` with a CLI argument.

Reply:
We've discussed this in a couple meetings so far. The current plan is to allow inlining extension configs exclusively through the recently GA'ed externalInputs feature. This will come in a future update.

The latest version of this PR was updated to only support supplying extension configurations via a bicepparam or parameters JSON file.